docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,420 @@
+---
+title: Unit testing and mocking with the Azure SDK for .NET
```

Comment:
Should we reference this older post on Mocking https://devblogs.microsoft.com/azure-sdk/unit-testing-and-mocking/ and only focus on the new and/or improved bits?

Reply:
My vote is no. Once this doc publishes, I'd like to do these 2 things:

1. Add a note at the top of that blog post that directs customers to the doc for the latest guidance.
2. Replace the content at https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Mocking.md with a link to the doc. The doc will be much more discoverable and localized.